eval.go

```diff
@@ -1091,14 +1091,32 @@ func (st *Runtime) evalBaseExpressionGroup(node Node) reflect.Value {
 			node.errorf("identifier %q is not available in the current scope %v", node, st.variables)
 		}
 
-		// limit the number of pointers to follow
-		for dereferenceLimit := 2; resolved.Kind() == reflect.Ptr && dereferenceLimit >= 0; dereferenceLimit-- {
-			if resolved.IsNil() {
+		// attempt dereferencing pointers to something printable
+		rTest := resolved
+		for dereferenceLimit := 2; rTest.Kind() == reflect.Ptr; dereferenceLimit-- {
+
+			if rTest.IsNil() {
 				return reflect.ValueOf("")
 			}
-			resolved = reflect.Indirect(resolved)
+
+			rTest = reflect.Indirect(rTest)
+
+			switch rTest.Kind() {
+			case reflect.String,
+				reflect.Bool,
+				reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64,
+				reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64,
+				reflect.Float32, reflect.Float64:
+				return rTest
+			}
+
+			if dereferenceLimit == 0 {
+				return reflect.ValueOf("<nil>")
```

Comment:
doesn't this make it impossible to ever do anything at all with, say, a `***MyStruct`?

I mean, hopefully never needs to have three levels of indirection, but we shouldn't make it impossible to work with those. I'd say we return the original value in this case. This function is not only used to render values into the output string, but also to evaluate the base expression in chain expressions, which is usually an identifier, meaning this code will run when evaluating `{{a.Name()}}` where `a` might be a pointer to a variable of `type A **Foo` or something like that.